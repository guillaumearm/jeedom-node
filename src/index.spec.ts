import { setupRecorder } from 'nock-record'
import Jeedom from '.'

const HOST = 'http://192.168.1.60'

const recordAndMatchSnapshot = (
  testName: string,
  recorder: (fileName: string) => Promise<{ completeRecording: () => void }>,
  runEffect: () => Promise<any>,
) => {
  test(testName, async () => {
    const { completeRecording } = await recorder(`jeedom-${testName}`)
    const result = await runEffect()
    completeRecording()
    expect(result).toMatchSnapshot()
  })
}

describe('jeedom-node', () => {
  describe('exports', () => {
    test('default export should be a function', () => {
      expect(typeof Jeedom).toBe('function')
    })
  })

  describe('api', () => {
    const recorder = setupRecorder({ mode: 'record' })

    const record = (testName: string, runEffect: () => Promise<any>) =>
      recordAndMatchSnapshot(testName, recorder, runEffect)

    const apikey = process.env.JEEDOM_API_KEY || ''
    const api = Jeedom({ host: HOST, apikey })

    record('ping', () => api.ping())
    record('version', () => api.version())
    record('datetime', () => api.datetime())

    describe('config', () => {
      record('config::byKey', () => api.config.byKey({ key: 'version' }))
    })
  })
})
