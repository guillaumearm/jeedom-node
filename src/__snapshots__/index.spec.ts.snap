// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jeedom-node api config config::byKey 1`] = `"3.2.11"`;

exports[`jeedom-node api datetime 1`] = `1541261780.4365`;

exports[`jeedom-node api ping 1`] = `"pong"`;

exports[`jeedom-node api version 1`] = `"3.2.11"`;
